ksqldb-functional-tests/src/test/resources/query-validation-tests/fk-join.json

```diff
@@ -0,0 +1,136 @@
+      "statements": [
+        "CREATE TABLE left_table (id1 BIGINT PRIMARY KEY, f1 BIGINT) WITH (kafka_topic='left_topic', format='JSON');",
+        "CREATE TABLE right_table (id2 BIGINT PRIMARY KEY, f2 BIGINT) WITH (kafka_topic='right_topic', format='JSON');",
+        "CREATE TABLE output AS SELECT id1, id2, f1, f2 FROM left_table AS lt JOIN right_table AS rt ON lt.id1 = rt.f2;"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "Invalid join condition: table-table joins require to join on the primary key of the right input table."
+      }
+    },
+    {
+      "name": "Should fail on n-way join (fk outer)",
+      "comments": [
+        "After we add support for binary FK-joins, this test should still fail, but with a different error:",
+        "The new error should point out that FK-joins cannot be used in n-way joins yet.",
+        "Note: this case should be supported after we lift the general restriction to not support FK-joins in n-way joins."
+      ],
+      "statements": [
+        "CREATE TABLE left_table (id1 BIGINT PRIMARY KEY, f1 BIGINT) WITH (kafka_topic='left_topic', format='JSON');",
+        "CREATE TABLE middle_table (id2 BIGINT PRIMARY KEY, f2 BIGINT) WITH (kafka_topic='middle_topic', format='JSON');",
+        "CREATE TABLE right_table (id3 BIGINT PRIMARY KEY, f3 BIGINT) WITH (kafka_topic='right_topic', format='JSON');",
+        "CREATE TABLE output AS SELECT id1, id2, id3, f1, f2, f3 FROM left_table JOIN middle_table ON id1 = id2 JOIN right_table ON f1 = id3;"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "Invalid join condition: foreign-key table-table joins are not supported."
+      }
+    },
+    {
+      "name": "Should fail on n-way join (fk inner)",
+      "comments": [
+        "After we add support for binary FK-joins, this test should still fail, but with a different error:",
+        "The new error should point out that FK-joins cannot be used in n-way joins yet.",
+        "Note: this case should be supported after we lift the general restriction to not support FK-joins in n-way joins."
+      ],
+      "statements": [
+        "CREATE TABLE left_table (id1 BIGINT PRIMARY KEY, f1 BIGINT) WITH (kafka_topic='left_topic', format='JSON');",
+        "CREATE TABLE middle_table (id2 BIGINT PRIMARY KEY, f2 BIGINT) WITH (kafka_topic='middle_topic', format='JSON');",
+        "CREATE TABLE right_table (id3 BIGINT PRIMARY KEY, f3 BIGINT) WITH (kafka_topic='right_topic', format='JSON');",
+        "CREATE TABLE output AS SELECT id1, id2, id3, f1, f2, f3 FROM left_table JOIN middle_table ON f1 = id2 JOIN right_table ON id2 = id3;"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "Invalid join condition: foreign-key table-table joins are not supported."
+      }
+    }
+  ]
+}
```

Comment:
Thanks for the test coverage! Can we verify that we either already have tests for, or add tests for the following:
- ambiguous join attribute
- invalid data source qualifier
- invalid data source alias
- positive test cases for qualifiers and aliases (hopefully these already exist)

Reply:
Seems there is only tests for "field does not exist" in `joins.json` -- happy to add those test, but seems orthogonal to this PR to close test gaps?

Reply:
Agreed that these tests should have already existed before this PR, but the fact that this PR changes logic that could impact these test cases means I'd feel more confident in the correctness of this PR if we had the test coverage. Feel free to add additional test coverage in a follow-up PR -- in general, these types of requests are not blocking, especially if there is other work blocked behind the functionality of the PR (which there is in this case).